src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -169,9 +169,7 @@ internal Metadata GetMetadata(string fullPath, DateTime snapshotTimestamp)
 
                     var storagesArray = storages.ToImmutable();
 
-                    var valueSource = _configurationService.Options.DisableReferenceManagerRecoverableMetadata
-                        ? ValueSource.Constant(newMetadata)
-                        : new RecoverableMetadataValueSource(newMetadata, storagesArray);
+                    var valueSource = ValueSource.Constant(newMetadata);
```

Comment:
because all usages of value-sources in this area are now 'constant-value-sources' (e.g. a strong reference in memory) we can greatly simplify things further.  That happens in following commit.